nipype/interfaces/cat12/preprocess.py

```diff
@@ -0,0 +1,363 @@
+    label_roi = File(exists=True, desc="Files with thickness values of ROI.")
+
+    ##########################################
+    # MRI .nii files
+    ##########################################
+
+    mri_images = List(File(exists=True), desc="Different segmented images.")
+
+    # Grey Matter
+    gm_modulated_image = File(exists=True, desc="Grey matter modulated image.")
+    gm_dartel_image = File(exists=True, desc="Grey matter dartel image.")
+    gm_native_image = File(exists=True, desc="Grey matter native space.")
+
+    # White Matter
+    wm_modulated_image = File(exists=True, desc="White matter modulated image.")
+    wm_dartel_image = File(exists=True, desc="White matter dartel image.")
+    wm_native_image = File(exists=True, desc="White matter in native space.")
+
+    # CSF
+    csf_modulated_image = File(exists=True, desc="CSF modulated image.")
+    csf_dartel_image = File(exists=True, desc="CSF dartel image.")
+    csf_native_image = File(exists=True, desc="CSF in native space.")
+
+    bias_corrected_image = File(exists=True, desc="Bias corrected image")
+    ##########################################
+    # Surface files
+    ##########################################
+
+    surface_files = List(File(exists=True), desc="Surface files")
+
+    # Right hemisphere
+    rh_central_surface = File(exists=True, desc="Central right hemisphere files")
+    rh_sphere_surface = File(exists=True, desc="Sphere right hemisphere files")
+
+    # Left hemisphere
+    lh_central_surface = File(exists=True, desc="Central left hemisphere files")
+    lh_sphere_surface = File(exists=True, desc="Sphere left hemisphere files")
+
+    # Report files
+    report_files = List(File(exists=True), desc="Report files.")
+    report = File(exists=True, desc="Report file.")
+
+
+class CAT12Segment(SPMCommand):
+    input_spec = CAT12SegmentInputSpec
+    output_spec = CAT12SegmentOutputSpec
+
+    def __init__(self, **inputs):
+        _local_version = SPMCommand().version
+        if _local_version and "12." in _local_version:
+            self._jobtype = "tools"
+            self._jobname = "cat.estwrite"
+
+        SPMCommand.__init__(self, **inputs)
+
+    def _format_arg(self, opt, spec, val):
+        """Convert input to appropriate format for spm
+        """
+        if opt in ["in_files"]:
+            if isinstance(val, list):
+                return scans_for_fnames(val)
+            else:
+                return scans_for_fname(val)
+        return super(CAT12Segment, self)._format_arg(opt, spec, val)
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        f = self.inputs.in_files[0]
+        pth, base, ext = split_filename(f)
+
+        outputs["mri_images"] = [os.path.join(os.path.join(pth, "mri"), f) for f in
+                                 os.listdir(os.path.join(pth, "mri"))
+                                 if os.path.isfile(os.path.join(os.path.join(pth, "mri"), f))]
```

Comment:
`os.path.join()` takes any number of arguments.

```suggestion
        outputs["mri_images"] = [os.path.join(pth, "mri", f) for f in
                                 os.listdir(os.path.join(pth, "mri"))
                                 if os.path.isfile(os.path.join(pth, "mri", f))]
```

But you can actually do this more succinctly with the `pathlib` library:

```suggestion
        outputs["mri_images"] = [str(mri) for mri in Path(pth).glob("mri/*") if mri.is_file()]
```

For the remaining examples, I'll use that idiom.